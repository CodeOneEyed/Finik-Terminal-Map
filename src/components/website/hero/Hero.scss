.heroVideo {
  mask-image: url('../../../assets//images/png/Kyrgyzstan.png');
  mask-repeat: no-repeat;
  mask-size: 100%;
  max-width: 100%;
}
