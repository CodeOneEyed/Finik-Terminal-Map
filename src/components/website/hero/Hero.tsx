/* External dependencies */
import { Div, Text, Image, Container } from 'atomize';
import React from 'react';
import { useTranslation } from 'react-i18next';

/* Local dependencies */
import StoreAndWebRedirect from './storeAndWebRedirect/StoreAndWebRedirect';

import mapVideo from '../../../assets/images/png/gif.mp4';
import FinikLogo from '../../../assets/images/svg/FinikLogo.svg';
import webIcon from '../../../assets/images/svg/webApp.svg';
import appStoreIcon from '../../../assets/images/svg/apple.svg';
import googlePlayicon from '../../../assets/images/svg/googlePlay.svg';

import { descrStyles, mainTitleStyles } from '../../../assets/styles/globalStyles';
import './Hero.scss';

export default function Hero() {
  const { t } = useTranslation();

  const heroContentStyles = {
    d: 'flex',
    flexDir: 'column',
    align: 'center',
    pos: 'absolute',
    left: '35%',
    top: '35%',
  };

  return (
    <Div bg="black">
      <Div d="flex" justify="center" pos="relative">
        <video className="heroVideo" muted loop autoPlay>
          <source src={mapVideo} />
        </video>
      </Div>
      <Div {...heroContentStyles}>
        <Div d="flex" align="center" m={{ b: '20px' }}>
          <Image w="60px" h="60px" src={FinikLogo} />
          <Text {...mainTitleStyles}>{t('map')}</Text>
        </Div>
        <Div m={{ b: '20px' }}>
          <Text {...descrStyles} textAlign="center">
            {t('finikDescr')}
          </Text>
        </Div>
        <Div d="flex" style={{ gap: 24 }}>
          <StoreAndWebRedirect icon={webIcon} descr={t('goToThe')} title={t('Web app')} />
          <StoreAndWebRedirect icon={googlePlayicon} descr={t('getItOn')} title="Google Play" />
          <StoreAndWebRedirect icon={appStoreIcon} descr={t('downloadOnThe')} title="App Store" />
        </Div>
      </Div>
    </Div>
  );
}
