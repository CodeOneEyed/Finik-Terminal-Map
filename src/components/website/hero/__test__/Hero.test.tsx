import React from 'react';
import { I18nextProvider } from 'gatsby-plugin-react-i18next';
import renderer from 'react-test-renderer';

import { configureTestStore } from '../../../../../tests/utils';
import i18n from '../../../../../tests/utils/i18nForTest';
import Hero from '../Hero';
import { Provider } from 'react-redux';

describe('AppFeatures component', () => {
  const store = configureTestStore();

  function getRenderComponent() {
    return (
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <Hero />
        </Provider>
      </I18nextProvider>
    );
  }
  it('should match the snapshot', () => {
    const component = renderer.create(getRenderComponent());

    expect(component.toJSON()).toMatchSnapshot();
  });
});
