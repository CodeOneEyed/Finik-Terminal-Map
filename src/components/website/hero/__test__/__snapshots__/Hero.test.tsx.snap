// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFeatures component should match the snapshot 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <video
      autoPlay={true}
      className="heroVideo"
      loop={true}
      muted={true}
    >
      <source
        src="test-file-stub"
      />
    </video>
  </div>
  <div
    className=""
  >
    <div
      className=""
    >
      <img
        className=""
        src="test-file-stub"
      />
      <p
        className=""
      >
        map
      </p>
    </div>
    <div
      className=""
    >
      <p
        className=""
      >
        finikDescr
      </p>
    </div>
    <div
      className=""
      style={
        {
          "gap": 24,
        }
      }
    >
      <a>
        <div
          className=""
        >
          <div
            className=""
          >
            <img
              className=""
              src="test-file-stub"
            />
          </div>
          <div
            className=""
          >
            <p
              className=""
            >
              goToThe
            </p>
            <p
              className=""
            >
              Web app
            </p>
          </div>
        </div>
      </a>
      <a>
        <div
          className=""
        >
          <div
            className=""
          >
            <img
              className=""
              src="test-file-stub"
            />
          </div>
          <div
            className=""
          >
            <p
              className=""
            >
              getItOn
            </p>
            <p
              className=""
            >
              Google Play
            </p>
          </div>
        </div>
      </a>
      <a>
        <div
          className=""
        >
          <div
            className=""
          >
            <img
              className=""
              src="test-file-stub"
            />
          </div>
          <div
            className=""
          >
            <p
              className=""
            >
              downloadOnThe
            </p>
            <p
              className=""
            >
              App Store
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
